TigerHacks-App/TigerHacks-App/Views/Resources/TigerTalksDetailViewController.swift

```diff
@@ -43,7 +43,26 @@ class TigerTalksDetailViewController: UIViewController {
     
     func getVideo(videoCode: String) {
         if let url = URL(string: "https://www.youtube.com/embed/\(videoCode)") {
+            
+            // Create the indicator
+            let activityIndicator = UIActivityIndicatorView(activityIndicatorStyle: .gray)
+            activityIndicator.hidesWhenStopped = true
+            view.addSubview(activityIndicator)
+            
+            // Position in center and start animating
+            activityIndicator.translatesAutoresizingMaskIntoConstraints = false
+            NSLayoutConstraint.activate([
+                activityIndicator.centerXAnchor.constraint(equalTo: videoWebView.centerXAnchor),
+                activityIndicator.centerYAnchor.constraint(equalTo: videoWebView.centerYAnchor)])
+            activityIndicator.startAnimating()
+            
+            // Load video
             videoWebView.load(URLRequest(url: url))
+            
+            // Stop the animation
+            DispatchQueue.main.asyncAfter(deadline: .now() + 2) {
```

Comment:
Instead of just stopping the animation after 2 seconds, we should make this variable, so it'll work well for any network speed. If we implement the ```WKNavigationDelegate```, we can implement this:

```swift
func webView(_ webView: WKWebView, didCommit navigation: WKNavigation!) {
        activityIndicator.stopAnimating()
}
```

Then you just start the animation, call the load, and it will stop the animation once its loaded. You also have to assign the ```videoWebView.delegate = self```